src/lib/FormGroupLayout.tsx

```diff
@@ -26,7 +29,7 @@ const FormGroupLayout = <T extends FieldValues>(props: FormGroupLayoutProps<T>)
   return (
     <FormGroup switch={switchLayout ? true : undefined} check={checkboxLayout ? true : undefined}>
       <Label check={checkboxLayout || switchLayout} for={id}>
-        {label}
+        {label} {requiredFields?.includes(name) && String(label)?.length > 0 && "*"}
```

Comment:
I suggest to `calculate` the label on line 25 and than here simply output the value. Makes it easier to read the code.